### PR TITLE
Maprenderer

### PR DESCRIPTION
**Pull request by deextercz**
This is part of @JeanCollas pull request where is modified maprenderer.
It is fixing bug with map in tabbed page.

### CODE REVIEW

**Comment by JeanCollas** [comment]
Maybe you could try to take the commits I did directly? :) (I know it can be done, I did it once ^^ cherry-picking commits, not very complicated but requires commands )
In that way merging the works could be easier and I could help! 
Otherwise I can split everything, but next week.

**Comment by deextercz** [comment]
You have a lot of commits so I think it is easier to replace all files with your files and repair errors than solve merge conflicts.
Also I will more understand your changes. But if you have easier / better solution I will follow your suggestions.

File: Xamarin.Forms.GoogleMaps/Xamarin.Forms.GoogleMaps/Interfaces/ICircle.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+        Color CircleFillColor { get; }
+
+        Position CircleCenter { get; }
+
+        Distance CircleRadius { get; }
+    }
+    public static class ICircleExtensions
+    {
+        public static Circle ToCircle(this ICircle iCircle)
```
Comment (deextercz) [inline Xamarin.Forms.GoogleMaps/Xamarin.Forms.GoogleMaps/Interfaces/ICircle.cs]: @amay077 could you take a look at this?

**Comment by amay077** [comment]
What is purpose of this PR?
If it is "fixing bug with map in tabbed page" then I think ``ICircle.cs`` is no need.

Please make 1PR = 1Purpose with minimum commits.
And make better title. 👍

**Comment by deextercz** [comment]
It seems that #101 is fixed

**Comment by deextercz** [comment]
@amay077 could you review and merge this branche? 
I need to solve conflicts everytime when someone is doing some changes.
Thank you.

**Comment by amay077** [comment]
I'll reject this PR.
Because:

* The purpose #101 does not fix completely
* Contains many **NON-RELATED** file changes (``IPin``, ``Map.PinsSource``, ``Delayer``, etc)
* I don't know why using ``Timer``

You should focus to only #101 and make file changes are only for #101 .

**Comment by deextercz** [comment]
I think I am not able to reset changes about PinSource rest of all non related changes were reseted.

**Comment by amay077** [comment]
I think you should re-start fix for #101 .

And I create a issue #196 .
How about Map.InitialCameraPosition?
Is this resolve your issue?

**Comment by deextercz** [comment]
@amay077 tadaaa it looks like this was enough to fix my issue, could you try it on iOS? on Android it works.

**Comment by amay077** [comment]
I released [1.8.0-beta5](https://www.nuget.org/packages/Xamarin.Forms.GoogleMaps/1.8.0-beta5) that contains ``Map.InitialCameraUpdate`` property.

I can not merge this PR because I don't have plan to add ``IPin``, ``MapMoveEnded``, ``PinsSource`` and more.
You can use your forked project if you needed but I can not try it.

**Comment by deextercz** [comment]
@amay077 thank you for your comment, but I wanted to solve #101 
It looks like the solution is to do this:

`

        public MapSpan VisibleRegion
        {
            get { return _visibleRegion; }
            internal set
            {
                if (_visibleRegion == value)
                    return;
                if (value == null)
                    throw new ArgumentNullException(nameof(value));
                OnPropertyChanging();
                _visibleRegion = value;
                OnPropertyChanged();
                InitialCameraUpdate = CameraUpdateFactory.NewBounds(value.ToBounds(), 0); # fix for #101
            }
        }
`

I don´t know the reason of using OnPropertyChanging(); and OnPropertyChanged(); maybe it could be removed.

**Pull request closed without merging**